src/coreclr/jit/gentree.cpp

```diff
@@ -18928,9 +18926,10 @@ CORINFO_CLASS_HANDLE Compiler::gtGetFieldClassHandle(CORINFO_FIELD_HANDLE fieldH
 #if DEBUG
             char fieldNameBuffer[128];
             char classNameBuffer[128];
-            JITDUMP("\nQuerying runtime about current class of field %s (declared as %s)\n",
-                    eeGetFieldName(fieldHnd, true, fieldNameBuffer, sizeof(fieldNameBuffer)),
-                    eeGetClassName(fieldClass, classNameBuffer, sizeof(classNameBuffer)));
+            eeGetFieldName(fieldHnd, true, fieldNameBuffer, sizeof(fieldNameBuffer));
+            eeGetClassName(fieldClass, classNameBuffer, sizeof(classNameBuffer));
```

Comment:
The idea of pulling these calls out of the JITDUMP is that we'll get the actual field and class names in SPMI collections, so they will replay with JitDump to give us the actual names.

Reply:
I think you should put it under `EnableExtraSuperPmiQueries` to avoid the extra metadata queries (even under debug). E.g. `pickGDV` does the same:
https://github.com/dotnet/runtime/blob/c52c808300dee381369b6301b876ee177765ce31/src/coreclr/jit/importercalls.cpp#L6247-L6306

Reply:
Good idea. Done.